onmt/transforms/misc.py

```diff
@@ -100,7 +113,8 @@ def warm_up(self, vocabs=None):
     def _prepend(self, example, prefix):
         """Prepend `prefix` to `tokens`."""
         for side, side_prefix in prefix.items():
-            example[side] = side_prefix.split() + example[side]
+            if example[side] is not None:
```

Comment:
yes it does always exist in our current case:
case 1: train/ valid dataset, pref dict is defined in _get_prefix()
case 2: inference: prefix_dict is defined in get_prefix_dict()
key exist even if opts.tgt_prefix does not exist.

Reply:
oh sorry I answered the wrong question

Reply:
But still the key 'tgt' of example always exist in our current master, but it can be None.
In essence for double security we can test if the key exists AND the value is not None.